terra_notebook_utils/blobstore/local.py

```diff
@@ -0,0 +1,117 @@
+import os
+import shutil
+from math import ceil
+from functools import wraps
+from typing import Dict, Generator, Optional
+
+from getm import default_chunk_size
+
+from terra_notebook_utils import blobstore
+
+
+def catch_blob_not_found(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except FileNotFoundError as ex:
+            raise blobstore.BlobNotFoundError(f"Could not find {self.url}") from ex
+    return wrapper
+
+class LocalBlobStore(blobstore.BlobStore):
```

Comment:
This may also want to optionally accept the `file://` schema.

Reply:
I prefer not to unless that use case crops up.